src/Yoti.Auth/Profile/BaseProfile.cs

```diff
@@ -150,24 +150,14 @@ public List<YotiAttribute<T>> FindAttributesStartingWith<T>(string prefix)
         {
             if (prefix == null)
                 throw new ArgumentNullException(nameof(prefix));
-
-            List<YotiAttribute<T>> matches = new List<YotiAttribute<T>>();
-
-            foreach (KeyValuePair<string, List<BaseAttribute>> attributesByName in _attributes)
-            {
-                if (attributesByName.Key.StartsWith(prefix, StringComparison.Ordinal))
-                {
-                    foreach (var attribute in attributesByName.Value)
-                    {
-                        if (attribute is YotiAttribute<T> castableAttribute)
-                        {
-                            matches.Add(castableAttribute);
-                        }
-                    }
-                }
-            }
-
-            return matches;
+            return 
+                _attributes.Where(
+                    attributesByName => attributesByName.Key.StartsWith(
+                        prefix, StringComparison.Ordinal))
+                .SelectMany(
```

Comment:
Seems condition 
`(attribute is YotiAttribute<T> castableAttribute)`
is not implemented here. 
If this changes the behaviour (?) we could fix with a suitable 
`.Where(a => a is YotiAttribute<T>)`